### DISCUSSION TITLE
GPL licence

### DISCUSSION THREAD

**Discussion by Stealth01** [microsoft/CsWin32#563] [General]
CsWin32 has the MIT licence, but it depends by the Windows.SDK that has its own licence.
I know the MIT licence is compatible with the GPL, but I don't know if the Windows.SDK licence too.
I am not an expert in licensing, so I ask those who know more about it:
Can I use the GPL licence for my application if it use CsWin32 and its dependencies?

**Comment by BenJKuhn** [comment]
Unfortunately, I can't provide legal advice. I can confirm that the CSWin32 tooling is distributed under the MIT license, and the metadata it's designed to consume (https://www.nuget.org/packages/Microsoft.Windows.SDK.Win32Metadata) is distributed using the Microsoft Windows SDK license (https://aka.ms/WinSDKLicenseURL). 

With regard to compatibility with the GPL, you'll want talk to legal counsel that can discuss your particular situation.

Thanks,

Ben

**Comment by AArnott** [comment]
Non-authoritative answer:
Keep in mind that you don't redistribute CsWin32. It is merely a build tool. The MIT license CsWin32 carries would apply to its own source code -- not the source code it generates. The source code it generates is intended for consumption by a project regardless of its license.
CsWin32 _may_ add a dependency on System.Memory to your application, but that's a standard .NET Framework or .NET library. 

As you seem to already have determined, the more relevant license for your concern is probably the one associated with the metadata. The metadata doesn't ship with your app either. But your app takes a dependency on Windows DLLs as a result of using CsWin32, just as it would if you had hand-written the interop. 
So maybe the key question is "can your app depend on Windows dlls?"

I'll see if we can get an authoritative answer added to our license or readme files.

**Comment by Stealth01** [comment]
Many thanks for the answer!

If I understand correctly, CsWin32 is a build tool just as Visual Studio is.

The GPL has an exception called "System Library Exception" that is explained here:
https://www.gnu.org/licenses/gpl-faq.en.html#SystemLibraryException
https://www.gnu.org/licenses/quick-guide-gplv3.en.html#less-source-to-distribute-new-system-libraries-exception
